Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/cparser/CParserTask.java

```diff
@@ -59,85 +58,62 @@ public CParserTask(CParserPlugin plugin, String[] filenames, String options,
 
 	@Override
 	public void run(TaskMonitor monitor) {
-		DataTypeManager fileDtMgr = null;
 		try {
-			if (dtMgr == null) {
+			if (dataFileName != null) {
 				File file = new File(dataFileName);
 				dtMgr = FileDataTypeManager.createFileArchive(file);
-				fileDtMgr = dtMgr;
 			}
 
 			plugin.parse(filenames, options, dtMgr, monitor);
+			if (monitor.isCancelled()) {
+				SwingUtilities.invokeLater(() -> {
+					Msg.showInfo(getClass(),
+						plugin.getDialog().getComponent(), "Cancelled", "Task was cancelled.");
+				});
+			}
+
 			if (dataFileName != null) {
 				if (dtMgr.getDataTypeCount(true) != 0) {
 					try {
 						((FileDataTypeManager) dtMgr).save();
-						dtMgr.close();
-						SwingUtilities.invokeLater(new Runnable() {
-							@Override
-							public void run() {
-								Msg.showInfo(
-									getClass(), plugin.getDialog().getComponent(),
-									"Created Archive File", "Successfully created archive file\n" +
-										((FileDataTypeManager) dtMgr).getFilename());
-							}
-
+						SwingUtilities.invokeLater(() -> {
+							Msg.showInfo(getClass(),
+								plugin.getDialog().getComponent(),
+								"Created Archive File", "Successfully created archive file\n" +
+									((FileDataTypeManager) dtMgr).getFilename());
 						});
 					}
-					catch (DuplicateFileException e) {
-						Msg.showError(this, plugin.getDialog().getComponent(), "Error During Save",
-							e.getMessage());
-					}
 					catch (Exception e) {
 						Msg.showError(this, plugin.getDialog().getComponent(), "Error During Save",
 							"Could not save to file " + dataFileName, e);
 					}
 					finally {
-						if (dtMgr instanceof FileDataTypeManager) {
-							dtMgr.close();
-						}
+						dtMgr.close();
 					}
 				}
-				else {
-					SwingUtilities.invokeLater(new Runnable() {
-						@Override
-						public void run() {
-							Msg.showInfo(getClass(),
-								plugin.getDialog().getComponent(), "Parse Errors", "File was not created due to parse errors.");
-						}
+				else if (!monitor.isCancelled()) {
+					SwingUtilities.invokeLater(() -> {
+						Msg.showInfo(getClass(),
+							plugin.getDialog().getComponent(), "Parse Errors", "No datatypes were found");
 					});
 				}
 			}
 		}
-		catch (ghidra.app.util.cparser.C.ParseException e) {
+		catch (ghidra.app.util.cparser.C.ParseException | ghidra.app.util.cparser.CPP.ParseException e) {
```

Comment:
Are the full package names necessary?

Reply:
Well Java doesn't have import aliasing, so best we could do is import one and use a fully qualified name for the other, that'd be uglier IMHO.

Reply:
> Well Java doesn't have import aliasing, so best we could do is import one and use a fully qualified name for the other, that'd be uglier IMHO.

Oh, I didn't realize they had the same class name. I don't know if this can be done or not,  but is it possible to import `ghidra.app.util.cparser` and then reference them as `C.ParseException` and `CPP.ParseException` like you can with static imports? It would be less noisy,  but if it's not possible then it's not possible.

Reply:
Can't import packages, only types.